UPGRADE_GUIDE.md

```diff
@@ -2,9 +2,146 @@
+```
+
+New method:
+```csharp
+Client myClient = new Client(new ClientConfiguration("my_api_key"));
+```
+
+The `ClientConfiguration` class has a number of optional parameters that can be set to customize the behavior of the `Client` instance.
+
+```csharp
+Client myClient = new Client(new ClientConfiguration("my_api_key") {
+  ApiBase = "optional_api_base_override",
+  CustomHttpClient = myOptionalCustomHttpClient,
+  Timeout = TimeSpan.FromMilliseconds(myCustomTimeoutMilliseconds)
+});
+```
+
+## 5.0 Naming Conventions
+
+*Likelihood of Impact: **Medium***
+
+A number of classes and properties have been renamed or altered:
+
+- `Smartrate` is now `SmartRate`
+
+## 5.0 Exception Changes
+
+*Likelihood of Impact: **Medium***
+
+Some exception types have been consolidated or altered:
+
+- `UnexpectedHttpError` is now `UnknownHttpError`
+- `UnknownApiError` has been removed and replaced with `UnknownHttpError`
+- `ExternalApiError` no longer inherits from `ApiError` and instead inherits from `EasyPostError`
+  - Any EasyPost API failure will raise an `ApiError`-based exception
+  - Non-EasyPost API/HTTP failures (e.g. calls to Stripe) will raise an `ExternalApiError` exception
+- All `EasyPostError` exceptions now have a `PrettyPrint` getter method that returns a human-readable string representation of the error
+- An EasyPost API timeout will raise a `TimeoutError` exception rather than a `System.Threading.Tasks.TaskCanceledException`
+
+## 5.0 Cancellation Tokens
+
+*Likelihood of Impact: **Medium***
+
+All service functions that make HTTP requests now accept an optional `CancellationToken` parameter that can be used to cancel the request.
+
+## 5.0 Parameter Sets
+
+*Likelihood of Impact: **Medium***
+
+The Parameter objects introduced in [`v4.5.0`](CHANGELOG.md#v450-2023-03-22) have been moved out of beta. As a result, the classes are available in a different namespace.
+
+Old namespace:
+```csharp
+var parameters = new EasyPost.BetaFeatures.Parameters.Addresses.Create();
+```
+
+New namespace:
+```csharp
+var parameters = new EasyPost.Parameters.Address.Create();
+```
+
+Note that the namespaces have also changed from plural to singular (e.g. `Addresses` to `Address`, `Shipments` to `Shipment`) to better correlate with the service names on a `Client` instance.
+
+## 5.0 Drop RestSharp Dependency
+
+*Likelihood of Impact: **Low***
+
+The RestSharp dependency in this SDK has been dropped in favor of using the `System.Net.Http` and `Newtonsoft.Json` libraries directly.
+
+This should have no impact on the end-user experience of using this SDK, but if you were reliant on the RestSharp dependency in this SDK being transitively used for another aspect of your codebase, you will need to now install RestSharp directly.
+
+## 5.0 Dispose Pattern
+
+*Likelihood of Impact: **Low***
+
+The SDK has implemented the IDisposable interface for a number of classes, including `Client` and `ClientConfiguration` and several internal classes.
```

Comment:
issue: if this doesn't affect user code, it shouldn't be listed here, instead it should be listed in the CHANGELOG